### PR TITLE
VehicleStopMode extensions

### PR DESCRIPTION
**Pull request by schwepmo**
## Type of this PR 

- [ ] Bug fix
- [x] Enhancement

## Description

* Parking at parking areas is now possible when using SUMO as traffic simulator
* made `VehicleStopMode` available in `VehicleData`
* to realize this, `VehicleStopMode` was extracted from `VehicleStop` interaction
* found a bug where electric vehicles could only charge if NOT stopped, which was now properly inverted
* tidied up `UnitData` and `VehicleData` by reorganizing fields/getters and builder and using `ToStringBuilder` instead of string concatenations

## Issue(s) related to this PR

 * relates to internal issues 223 and 230
 
 
## Definition of Done

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] All commits are signed-off (`-s`) with your mail address of your Eclipse Account.
- [x] `origin/main` has been merged into your Fork.
- [x] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [x] There are no new SpotBugs warnings. 
- [x] Coding guidelines have been observed (see CONTRIBUTING.md).
- [x] All tests pass.

### CODE REVIEW

File: fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/ElectricVehicleUnit.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public BatteryData getBatteryState() {
 
     @Override
     public void sendChargingStartRequest(String chargingStationId) {
-        if (getVehicleData() != null && !getVehicleData().isStopped()) {
+        if (getVehicleData() != null && getVehicleData().isStopped()) {
```
Comment (schwepmo) [inline fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/ElectricVehicleUnit.java:86]: prior to this PR, this was actually wrong. But we could also consider just removing this if-clause

File: lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/vehicle/VehicleStopMode.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    PARKING_AREA;
+
+    /**
+     * Returns the corresponding integer for different stop modes according to
+     * <a href="https://sumo.dlr.de/docs/TraCI/Change_Vehicle_State.html#stop_0x12">stop</a>.
+     *
+     * @return the corresponding int to the stop mode
+     */
+    public int stopModeToInt() {
```
Comment (kschrab) [inline lib/mosaic-objects/src/main/java/org/eclipse/mosaic/lib/objects/vehicle/VehicleStopMode.java:48]: I'd rather move this method to mosaic-sumo, as this is SUMO specific code

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java [hunk 3]
```diff
@@ -371,16 +371,17 @@ public TraciSimulationStepResult simulateUntil(long time) throws InternalFederat
 
                 lastVehicleData = this.lastVehicleData.get(veh.id);
 
-                VehicleStop.VehicleStopMode vehicleStopMode = getStopMode(veh.stoppedStateEncoded);
-                if (vehicleStopMode == VehicleStop.VehicleStopMode.PARK) {
+                VehicleStopMode vehicleStopMode = getStopMode(veh.stoppedStateEncoded);
+                if (vehicleStopMode == VehicleStopMode.PARK) {
```
[the remark this answers is not in the document]
Comment (schwepmo) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java:375]: Actually yeah, that's why the order of the if-statement is important. Any suggestion for improvement?

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java [hunk 4]
```diff
@@ -396,7 +397,7 @@ public TraciSimulationStepResult simulateUntil(long time) throws InternalFederat
                             .orientation(DriveDirection.UNAVAILABLE, veh.heading, veh.slope)
                             .route(veh.routeId)
                             .signals(decodeVehicleSignals(veh.signalsEncoded))
-                            .stopped(vehicleStopMode != null)
+                            .stopped(vehicleStopMode != VehicleStopMode.NOT_STOPPED, vehicleStopMode)
```
[the remark this answers is not in the document]
Comment (schwepmo) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java:400]: The second parameter for the `stopped()`-method in the builder is the `VehicleStopMode`. I suppose we could omit the flag altogether. 🤷

[the remark this answers is not in the document]
Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java:401]: yes please do so

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java]: I don't get it :D

[the remark this answers is not in the document]
Comment (schwepmo) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/facades/SimulationFacade.java]: Oh I was looking at the wrong line of code 😅 You're right

**Review by kschrab — APPROVED** [review]

**Pull request merged**